tests/extension/oneapi_non_uniform_groups/opportunistic_group_api.cpp

```diff
@@ -0,0 +1,182 @@
+
+    sycl::nd_range<1> nd_range{sycl::range<1>{num_items},
+                               sycl::range<1>{num_items}};
+
+    cgh.parallel_for<struct OpportunisticGroupAPI>(
+        nd_range, [=](sycl::nd_item<1> it) {
+          auto& results = acc[it.get_global_id()];
+
+          size_t sub_group_size = it.get_sub_group().get_local_range().size();
+
+          auto opportunistic =
+              oneapi_ext::this_kernel::get_opportunistic_group();
+          static_assert(
+              std::is_same_v<decltype(opportunistic), opportunistic_group_t>);
+
+          results[checks::get_group_id] = opportunistic.get_group_id() == 0;
+          results[checks::get_local_id] =
+              opportunistic.get_local_id() <
+              opportunistic.get_local_range().size();
+          results[checks::get_group_range] =
+              opportunistic.get_group_range().size() == 1;
+          results[checks::get_local_range] =
+              opportunistic.get_local_range().size() <= sub_group_size;
+          results[checks::get_group_linear_id] =
+              opportunistic.get_group_linear_id() == 0;
+          results[checks::get_local_linear_id] =
+              opportunistic.get_local_linear_id() ==
+              opportunistic.get_local_id();
+          results[checks::get_group_linear_range] =
+              opportunistic.get_group_linear_range() ==
+              opportunistic.get_group_range().size();
+          results[checks::get_local_linear_range] =
+              opportunistic.get_local_linear_range() ==
+              opportunistic.get_local_range().size();
+          results[checks::leader] =
+              opportunistic.leader() == (opportunistic.get_local_id() == 0);
+        });
+  });
+  q.wait();
+
+  CheckResults results = {};
+  sycl::accessor acc = results_buffer.get_host_access();
+  for (size_t check = 0; check < checks::COUNT; check++)
+    results[check] = std::all_of(acc.cbegin(), acc.cend(),
+                                 [=](const auto& it) { return it[check]; });
+
+  // Group-category traits.
+  STATIC_CHECK(sycl::is_group<opportunistic_group_t>::value);
+  STATIC_CHECK(sycl::is_group_v<opportunistic_group_t>);
+  STATIC_CHECK(
+      oneapi_ext::is_user_constructed_group<opportunistic_group_t>::value);
+  STATIC_CHECK(oneapi_ext::is_user_constructed_group_v<opportunistic_group_t>);
+  STATIC_CHECK(
+      !oneapi_ext::is_fixed_topology_group<opportunistic_group_t>::value);
+  STATIC_CHECK(!oneapi_ext::is_fixed_topology_group_v<opportunistic_group_t>);
+
+  // Aliases.
+  STATIC_CHECK(std::is_same_v<opportunistic_group_t::id_type, sycl::id<1>>);
+  STATIC_CHECK(
+      std::is_same_v<opportunistic_group_t::range_type, sycl::range<1>>);
+  STATIC_CHECK(std::is_same_v<opportunistic_group_t::linear_id_type, uint32_t>);
+
+  // Static constexpr members.
+  STATIC_CHECK(opportunistic_group_t::dimensions == 1);
+  STATIC_CHECK(opportunistic_group_t::fence_scope ==
+               sycl::sub_group::fence_scope);
```

Comment:
Is `sycl::sub_group::fence_scope` the same as `sycl::memory_scope::sub_group`?

Reply:
They are the same, but you are right that we should check this based on the definition of the class, so I've changed it.